build-logic/binary-compatibility/src/main/groovy/gradlebuild/EnrichedReportRenderer.groovy

```diff
@@ -0,0 +1,104 @@
+     *
+     * Instead, we'll encode the path to the file in the description data field, as a link.  This is
+     * useful regardless of this renderer's needs, since now there will be a link embedded in the report
+     * to quickly open the changes file.  By then we can also regex that path out of the description
+     * in order to create a changes {@link File} for further use in this class.
+     *
+     * @param data the report data containing a description to parse
+     * @return accepted api changes reported upon, as a file
+     */
+    private static File getAcceptedApiChangesFile(RichReportData data) {
+        def matcher = data.description =~ acceptedChangesRegex
+        return new File(matcher[0][1])
+    }
+
+    private static RichReportData enrichReport(RichReportData data) {
+        String currentApiChanges = getAcceptedApiChangesFile(data).text
+        String enrichedDesc = data.description + buildFixAllButton(currentApiChanges)
+        return new RichReportData(data.reportTitle, enrichedDesc, data.violations)
+    }
+
+    private static String buildFixAllButton(String currentApiChanges) {
+        return """
+            <script type="text/javascript">
+                function getAllErrorCorrections() {
+                    var changeElements = \$(".well pre")
+                    var result = []
+                    changeElements.each(function() {
+                        result.push(JSON.parse(this.textContent));
+                    })
+                    return result;
+                }
+
+                function appendErrorCorrections() {
+                    var result = $currentApiChanges;
+                    getAllErrorCorrections().forEach(function(correction) {
+                        result.acceptedApiChanges.push(correction);
+                    });
+                    result.acceptedApiChanges = result.acceptedApiChanges.sort(function(a, b) {
+                        return (a.type + a.member) - (b.type + b.member);
+                    });
+                    return result;
+                }
+
+                function acceptAllErrorCorrections() {
+                    var textToWrite = JSON.stringify(appendErrorCorrections(), null, 4) + "\\n";
+                    var textFileAsBlob = new Blob([textToWrite], {type:'text/plain'});
+                    var fileNameToSaveAs = 'accepted-public-api-changes.json';
+
+                    var downloadLink = document.createElement("a");
+                    downloadLink.download = fileNameToSaveAs;
+                    downloadLink.innerHTML = "Download File";
+
+                    if (window.webkitURL != null) {
+                        // Chrome allows the link to be clicked without actually adding it to the DOM
+                        downloadLink.href = window.webkitURL.createObjectURL(textFileAsBlob);
+                    } else {
+                        // Firefox requires the link to be added to the DOM before it can be clicked
```

Comment:
Does the firefox method work for chrome too? Why not just include one solution if it works for both browsers?